src/TTImp/ProcessDef.idr

```diff
@@ -441,13 +441,12 @@ checkClause mult vis totreq hashit n opts nest env (ImpossibleClause fc lhs)
 checkClause {vars} mult vis totreq hashit n opts nest env (PatClause fc lhs_in rhs)
     = do (_, (vars'  ** (sub', env', nest', lhstm', lhsty'))) <-
              checkLHS False mult hashit n opts nest env fc lhs_in
-         let rhsMode = if isErased mult then InType else InExpr
          log "declare.def.clause" 5 $ "Checking RHS " ++ show rhs
          logEnv "declare.def.clause" 5 "In env" env'
 
          rhstm <- logTime ("+++ Check RHS " ++ show fc) $
                     wrapErrorC opts (InRHS fc !(getFullName (Resolved n))) $
-                       checkTermSub n rhsMode opts nest' env' env sub' rhs (gnf env' lhsty')
+                       checkTermSub n (InExpr mult) opts nest' env' env sub' rhs (gnf env' lhsty')
```

Comment:
This is one of the InExpr (RHS Expr) that was converted to an InType when Rig0 /cc @gallais